tests/admin/valuesets/valueset-creation.spec.ts

```diff
@@ -0,0 +1,380 @@
+    await page.getByRole("button", { name: "+ Add Concepts" }).first().click();
+    await page.getByRole("textbox", { name: "Code *" }).first().fill("1558-6");
+
+    await page.getByRole("button", { name: "+ Add Concepts" }).first().click();
+    const secondConcept = page.getByRole("textbox", { name: "Code *" }).nth(1);
+    await secondConcept.fill("2339-0");
+
+    // Save and verify
+    await page.getByRole("button", { name: "Save" }).click();
+
+    await expect(page.getByText("ValueSet created successfully")).toBeVisible({
+      timeout: 10000,
+    });
+    await page.waitForLoadState("networkidle");
+    await expect(page.getByRole("cell", { name: uniqueName })).toBeVisible();
+  });
+
+  test("should create valueset with include rules - any system with concept only", async ({
+    page,
+  }) => {
+    const uniqueId = crypto.randomUUID().slice(0, 8);
+    const uniqueName = `Concept Only ValueSet ${uniqueId}`;
+    const uniqueSlug = `concept-only-${uniqueId}`;
+
+    // Fill basic info
+    await page.getByRole("textbox", { name: "Name *" }).fill(uniqueName);
+    await page.getByRole("textbox", { name: "Slug *" }).fill(uniqueSlug);
+
+    // Add include rule with SNOMED concept
+    await page
+      .getByRole("button", { name: "+ Add Rule", exact: true })
+      .first()
+      .click();
+    await page.getByRole("combobox", { name: "System" }).first().click();
+    await page.getByRole("option", { name: "SNOMED CT" }).click();
+
+    // Add concept and verify it's properly added
+    await page.getByRole("button", { name: "+ Add Concepts" }).first().click();
+    const conceptCode = "73211009"; // Diabetes mellitus SNOMED code
+    await page
+      .getByRole("textbox", { name: "Code *" })
+      .first()
+      .fill(conceptCode);
+
+    // Set status
+    await page.getByRole("combobox", { name: "Status" }).click();
+    await page.getByRole("option", { name: "Active" }).click();
+
+    // Verify concept is properly set before saving
+    await expect(
+      page.getByRole("textbox", { name: "Code *" }).first(),
+    ).toHaveValue(conceptCode);
+
+    // Test ValueSet Preview
+    await page.getByRole("button", { name: "Value Preview" }).click();
+
+    // Verify preview dialog opened
+    const previewDialog =
+      page.getByRole("dialog").first() ||
+      page.locator('[data-testid="preview-dialog"]').first();
+    await expect(previewDialog).toBeVisible();
+
+    // Verify specific codes appear in preview
+    await expect(previewDialog.getByText("73211009")).toBeVisible();
+
+    // Close preview and save
+    await page
+      .getByRole("button", { name: /close|back/i })
+      .first()
+      .click();
+    await page.getByRole("button", { name: "Save" }).click();
+
+    // Verify success
+    await expect(page.getByText("ValueSet created successfully")).toBeVisible({
+      timeout: 10000,
+    });
+  });
+
+  test("should create valueset with include rules - any system with filters", async ({
```

Comment:
- add test for each filter operation available for this value and verify there are only visible preview page